### ISSUE TITLE
ResizeOptions is not working in Fresco 1.7.1？

### ISSUE THREAD

**Issue by MidoriInu1** [facebook/fresco#2000]
#### Description
Excuse me.
I can not confirm this is a bug, but when I update Fresco version to 1.7.1, ResizeOptions is not working, but it works normally in version 1.5.0.
Please help me to test this function.
thx.

##### This is my code:
``` java
public static void loadImgByFile(CommonImageView imageView, String path) {
        if (imageView != null && !TextUtils.isEmpty(path)) {
            File file = new File(path);
            //if size too large then resize
            if (file.exists()) {
                BitmapFactory.Options options = new BitmapFactory.Options();
                options.inJustDecodeBounds = true;
                BitmapFactory.decodeFile(path, options);
                int imageHeight = options.outHeight;
                int imageWidth = options.outWidth;
                int maxSize = getGLESTextureMaxSize();
                Log.e("texture---test---->", "imageHeight=" + imageHeight
                        + "    imageWidth=" + imageWidth
                        + "    maxSize=" + maxSize
                );
                if (maxSize <= imageHeight || maxSize <= imageWidth) {
                    ImageRequest request = ImageRequestBuilder.newBuilderWithSource(Uri.fromFile(file))
                            .setResizeOptions(new ResizeOptions(maxSize/2, maxSize/2,maxSize))
                            .build();

                    DraweeController controller = Fresco.newDraweeControllerBuilder()
                            .setImageRequest(request)
                            .setOldController(imageView.getController())
                            .build();
                    imageView.setController(controller);
                    Log.e("texture---test---->", "controller=" + controller);
                    return;
                }
            }
            imageView.setImageURI(Uri.parse("file://" + path));
        }
    }
```

```java
private static int OPENGL_MAX_SIZE = 0;
    private static int getGLESTextureMaxSize() {
        if (OPENGL_MAX_SIZE != 0) {
            return OPENGL_MAX_SIZE;
        }
        EGL10 egl = (EGL10) EGLContext.getEGL();
        EGLDisplay dpy = egl.eglGetDisplay(EGL10.EGL_DEFAULT_DISPLAY);
        int[] vers = new int[2];
        egl.eglInitialize(dpy, vers);
        int[] configAttr = {
                EGL10.EGL_COLOR_BUFFER_TYPE, EGL10.EGL_RGB_BUFFER,
                EGL10.EGL_LEVEL, 0,
                EGL10.EGL_SURFACE_TYPE, EGL10.EGL_PBUFFER_BIT,
                EGL10.EGL_NONE
        };
        EGLConfig[] configs = new EGLConfig[1];
        int[] numConfig = new int[1];
        egl.eglChooseConfig(dpy, configAttr, configs, 1, numConfig);
        if (numConfig[0] == 0) {// TROUBLE! No config found.
        }
        EGLConfig config = configs[0];
        int[] surfAttr = {
                EGL10.EGL_WIDTH, 64,
                EGL10.EGL_HEIGHT, 64,
                EGL10.EGL_NONE
        };
        EGLSurface surf = egl.eglCreatePbufferSurface(dpy, config, surfAttr);
        final int EGL_CONTEXT_CLIENT_VERSION = 0x3098; // missing in EGL10
        int[] ctxAttrib = {
                EGL_CONTEXT_CLIENT_VERSION, 1,
                EGL10.EGL_NONE
        };
        EGLContext ctx = egl.eglCreateContext(dpy, config, EGL10.EGL_NO_CONTEXT, ctxAttrib);
        egl.eglMakeCurrent(dpy, surf, surf, ctx);
        int[] maxSize = new int[1];
        GLES10.glGetIntegerv(GLES10.GL_MAX_TEXTURE_SIZE, maxSize, 0);
        egl.eglMakeCurrent(dpy, EGL10.EGL_NO_SURFACE, EGL10.EGL_NO_SURFACE,
                EGL10.EGL_NO_CONTEXT);
        egl.eglDestroySurface(dpy, surf);
        egl.eglDestroyContext(dpy, ctx);
        egl.eglTerminate(dpy);
        OPENGL_MAX_SIZE = maxSize[0];
        return OPENGL_MAX_SIZE;
    }
```

##### When i run them in Fresco 1.7.1 ,the Logcat showing:
![image](https://user-images.githubusercontent.com/11972411/34199347-728b3488-e5a8-11e7-91d8-c93cebf50d68.png)

The ResizeOptions is not working.
SimpleDraweeView shows blank.
I think it is because that OpenGLRenderer can't load the jpeg image, when the height or width of the jpeg image is too large. DraweeController should have loaded the ResizeOptions which I set, but it uploaded the original image to OpenGLRenderer instead.

##### When i run them in  Fresco 1.5.0 ,the Logcat showing:
![image](https://user-images.githubusercontent.com/11972411/34199811-ecf7642a-e5a9-11e7-9d66-d5f81a113671.png)
The ResizeOptions is working normally.
SimpleDraweeView can show pictures.

#### Just a guess
I noticed in the page：
![image](https://user-images.githubusercontent.com/11972411/34200238-48be38c8-e5ab-11e7-9d68-99f2abc6407f.png)
Region decoding support is added in this version, this commit(bd249fa) changed ArtDecoder.
I can not confirm the problem because of this commit.

#### Some device parameters 
+ Device Model:  HUAWEI P9 (HUAWEI EVA-AL00)
+ Platform:  Android 7.0 
+ API level:  24
+ Screen resolution:  1080*1920
+ CPU:  Hisilicon Kirin 955 
+ RAM:  3.0 GB

**Comment by dmitry-voronkevich** [comment]
Thanks for reporting this issue. Maybe, @oprisnik can check if this issue caused by region decoding.

**Comment by erikandre** [comment]
Hi @MidoriInu1 

Thanks for submitting the bug report. I tried to reproduce this in the Showcase app using our normal test images, but no luck.

Would it be possible for you to share one of your test images so that we can try it out? Thanks!

**Comment by MidoriInu1** [comment]
Hi @erikandre 
I uploaded a zip file containing three of the test images. 
[images.zip](http://6d0dbc33a6c02.cdn.sohucs.com/foobar/images.zip)
Please download and unzip the file, then put the images into SD card, and call setImageURI(Uri.fromFile(file)).

Those images are test cases of my project, to test corner situations. I can reproduce the issue with those images in Fresco 1.7.1.

Hope this post may help, thx!

**Comment by AlexCJW** [comment]
I met this issue, too. I have to roll back fresco in my project to 1.5.0.

**Comment by jyh149129** [comment]
me too

**Comment by eaggle** [comment]
me too

**Comment by oprisnik** [comment]
I've just tested with the current master version and the Showcase app correctly resizes JPEG images. However, you're right @MidoriInu1, PNG downsampling is broken. I should have a fix shortly. Thanks for reporting!

**Comment by eaggle** [comment]
webp have the same problem

**Comment by oprisnik** [comment]
The fix is now available, see e2c464efbc9214d1a00aa416d3d9cfb26c0298da. Also fixes the issue for WebP.
